client/python/gradio_client/client.py

```diff
@@ -496,6 +503,14 @@ def submit(
             >> 9.0
         """
         inferred_fn_index = self._infer_fn_index(api_name, fn_index)
+        cancellable = False
+        cancel_fn_index = None
+        for i, dep in enumerate(self.config["dependencies"]):
+            if inferred_fn_index in dep["cancels"]:
+                cancellable = True
+                cancel_fn_index = i
```

Comment:
I don't think this will work if we have a single event that cancels multiple events. e.g. something like:

```py
import gradio as gr

with gr.Blocks() as demo:
    t1 = gr.Textbox()
    t2 = gr.Textbox()
    t3 = gr.Textbox()   
    t4 = gr.Textbox()
    e = t1.change(lambda x:x, t1, t2)
    f = t2.change(lambda x:x, t2, t3)
    t4.change(lambda x:x, t4, t3, cancels=[e, f])
```

if I understand correctly, let's say a user starts, via api, jobs e and f. If they cancel one of them, it'll cancel the other as well

Reply:
I think this is actually ok because when using the app via the UI, the `t4.change` event will also cancel both events. So I don't think the client should do something different from the UI.

That being said, I will look into your suggestion of creating a unique cancel event for every dependency so that every event can be cancelled from a client going forward!

Reply:
So I would agree that if someone calls the route corresponding to the `t4.change()` event, they may expect `e` and `f` to get canceled.

But if someone starts the `e` job on its own via the api, and they cancel it using `job_e.cancel()`, I don't think the expectation is that `job_f` should get canceled as well